javascript/ql/src/semmle/javascript/frameworks/AsyncPackage.qll

```diff
@@ -0,0 +1,218 @@
+/**
+ * Provides classes for working with [async](https://www.npmjs.com/package/async).
+ */
+import javascript
+
+module AsyncPackage {
+  /**
+   * Gets a reference the given member of the `async` or `async-es` package.
+   */
+  pragma[noopt]
```

Comment:
The join ordering was really weird; it looked like a bug in the optimizer. I talked to Alex ET about it, I'll follow up and open an issue (can't ping him here).

Basically the join ordering we got was to scan the `SourceNode` relation twice, each time joining it with `moduleMember`. In the `memberVariant` predicate below, it would then check again that the result of `member()` was still a `SourceNode`.

Reply:
The fixes are in flight apparently,

> it should be fixed by  #28687 and #28665

I'd like to leave `pragma[noopt]` for now so we can safely land on `master`.